### DISCUSSION TITLE
Azure Credits for Extension Authors

### DISCUSSION THREAD

**Discussion by isidorn** [microsoft/vscode-discussions#135] [Announcements]
We have 2000$ worth of Azure Credits to donate per VS Code Extension Author 🎉  

If you are interested please comment here and quickly explain what is your extension and how would you use these credits or send me an email inikolic@microsoft.com

We believe these credits can help you test, validate and publish your open source extension to the Marketplace in a secure way from an Azure Pipeline 🚀  
We do the same for the VS Code extensions we own and I can offer some best practice advice if needed.

We expect to give out these credits end of October, and based on how this process goes we might repeat it in the future. The amount will most likely be 2000$ per extension. Feel free to also nominate some extension you do not own.

This initiative is part of the Azure Credits for Open Source program https://opensource.microsoft.com/azure-credits

Thanks 🙏

**Comment by estruyf** [comment]
This is a great initiative @isidorn from you and your team. I'm sure this will help many developers and extensions.

I just have a couple of questions:

- Can the credits be used to run production services, or is it only for testing purposes? For instance, an API, website (documentation), ...
- If running production services is allowed. What will the renewal procedure (after a year) look like? Just want to make sure that services are not stopped or unexpected bills appear.

**Reply by isidorn** [reply]
@estruyf thanks for the feedback. Let me answer:
* Yes, you can use the credits as you like and we would love if this is tied to your extension. For example if your open source extension has a website or relies on a service it is perfectly fine to use these credits to run that production website / service.
* The same author can get the credits multiple times if they reapply in the future. However I can not make any long term future guarantees as we are just setting up this program for extension authors.

**Reply by estruyf** [reply]
Thanks for the quick answer @isidorn - the website/APIs would indeed be linked to the extension like for a documentation website, or APIs that extension uses.

**Reply by jeffwilcox** [reply]
The renewal question is key - this is a pilot program. While we're able to make these one-time grants, we aren't able to make any long-term commitment, and we'd hate for anyone to be surprised when these credits expire. This is one reason it's probably better for prototyping work, validating and testing with higher-end premium products in Azure, as opposed to running a production service to back the extension unfortunately.

**Comment by alefragnani** [comment]
Hi @isidorn ,

Congratulations for such amazing initiative! It will be a great improvement for extension developers toolbox 👏 

Right out of my head I can see at least two features I could take advantage of: 
* Application Insights: to better understand how an extension is used
* Virtual Machines: to test and validate extensions on Windows and Linux machines

But, I'm pretty sure more could come in the future, like services, websites and Azure Pipelines, as you pointed out.

Thank you

**Reply by jeffwilcox** [reply]
Absolutely do agree on the VM use case - in the larger open source credits pilot program, we've helped connect many teams to large virtual machines with Windows SKUs to validate, build and test there.

**Comment by jonmikeli** [comment]
Great initiative @isidorn . Is there something similar with .NET and VS extensions?

**Reply by isidorn** [reply]
@jonmikeli I work on the VS Code extension eco system so I am not sure if there is something similar for Visual Studio extensions. 
I can double check, but my suspicion is that there is not a similar program for VS right now. That might change depending on how this pilot program goes for VS Code extensions.

**Comment by PEZ** [comment]
Awesome! I've been asking for something like this in the extension dev Slack channel a few times. My main initial uses would be similar to what @alefragnani describes. I repeat here:

* Application insights
* Windows compatibility

For mainly for these two of my extensions:

* Calva
* Joyride

**Comment by PEZ** [comment]
I want to express my gratitude for this way of sponsoring my work with VS Code extensions. ♥️🙏♥️

My main usage is to have a Windows machine to try out issues reported by Windows users, and for fixing those issues. As the virtual machine can pretty slow, I use the generous amount credits to procure a really beefy machine. It is worth tons for me to be able to do this.

Thanks!

**Reply by isidorn** [reply]
Thank you @pez 🙏 

fyi @jeffwilcox
